internal/pkg/mapper/template/replacekeys/mapper.go

```diff
@@ -38,7 +40,16 @@ func (m *replaceKeysMapper) OnRemoteChange(changes *model.RemoteChanges) error {
 		if err := m.state.Set(modified); err != nil {
 			errors.Append(err)
 		}
+
+		replaced[original.Key().String()] = modified
 	}
+	
+	changes.Replace(func(v model.ObjectState) model.ObjectState {
+		if modified, found := replaced[v.Key().String()]; found {
+			return modified
+		}
+		return v
+	})
```

Comment:
Na predchadzajucich riadkoch sa zmaze stary ObjectState zo `state` a ulozi sa modifikovany.
Kedze sa zmenil `Key`, tak sa to musi tak robit, ... kedze objekty su ukladane podla key.

Tu pridavam este nahradenie za nove ObjectState v `RemoteChanges` strukture, aby ostatny kod fungoval spravne ... a `RemoteChanges` struktura mala rovnake objekty ako `state`.